### ISSUE TITLE
White screen

### ISSUE THREAD

**Issue by KewinOsborn** [ericferon/glpi-archibp#4]
Business Processes
GPL v2+
Eric Feron
2.0.5

when enabled all glpi is whitescreen. Nothing comes.

**Comment by ericferon** [comment]
Hi,
You should look for error messages into the logs.
Without this information, you cannot be helped.
K.r,

**Comment by ghost** [comment]
The same issue - white screen and glpi not working

**Comment by ericferon** [comment]
Hi,
Same issue, same answer : without precise information, I cannot help you ...

**Comment by Subtixx** [comment]
Possible problem:

> [Wed Oct 04 16:05:13.957866 2023] [php:warn] [pid 1101] [client 172.19.0.2:57036] PHP Warning:  Invalid relations declared between "glpi_documents" and "glpi_plugin_manageentities_companies" table. Target field "logo_id" is not a foreign key field. in /var/www/html/glpi/src/DbUtils.php on line 2176, referer: https://URL/front/marketplace.php
[Wed Oct 04 16:05:14.006446 2023] [php:warn] [pid 1101] [client 172.19.0.2:57036] PHP Warning:  Invalid relations declared for "glpi_plugin_accounts_accountstates" table. Target table "glpi_plugin_accounts_mailingstates" does not correspond to a known itemtype. in /var/www/html/glpi/src/DbUtils.php on line 2111, referer: https://URL/front/marketplace.php
[Wed Oct 04 16:05:14.036205 2023] [php:warn] [pid 1101] [client 172.19.0.2:57036] PHP Warning:  Invalid relations declared between "glpi_documents" and "glpi_plugin_manageentities_companies" table. Target field "logo_id" is not a foreign key field. in /var/www/html/glpi/src/DbUtils.php on line 2176, referer: https://URL/front/marketplace.php
[Wed Oct 04 16:05:14.040251 2023] [php:warn] [pid 1101] [client 172.19.0.2:57036] PHP Warning:  Invalid relations declared for "glpi_plugin_accounts_accountstates" table. Target table "glpi_plugin_accounts_mailingstates" does not correspond to a known itemtype. in /var/www/html/glpi/src/DbUtils.php on line 2111, referer: https://URL/front/marketplace.php

**Comment by ericferon** [comment]
Hi,
Thank you for the information.
In these error messages, I don't see any one related to archibp.
Maybe can you uninstall the plugin, see whether the symptom disappears, reinstall it and look for error messages if the symptom appears again.
K.r,

**Comment by Subtixx** [comment]
There is no other errors.. The page just stops rendering after the header. Disabling the plugin archibp makes the page work again.

**Comment by ericferon** [comment]
Hi,
There are probably error messages somewhere : either in <GLPI_DIR>/files/_log/php_errors.log, or in the php error log, on your web server (Apache, Nginx, ...).
K.r,

**Comment by daufveila** [comment]
Perhaps this will help in the search for the error:

This is what i found in `<GLPI_DIR>files/_log/php-errors.log`

```bash
[2023-12-06 13:58:55] glpiphplog.CRITICAL:   *** Uncaught Exception Error: Call to a member function fetch_assoc() on bool in /var/www/glpi/src/DBmysql.php at line 570
  Backtrace :
  marketplace/archibp/hook.php:236                   DBmysql->fetchAssoc()
  src/Plugin.php:1750                                plugin_archibp_getDropdown()
  src/Plugin.php:1765                                Plugin::doOneHook()
  src/Dropdown.php:1264                              Plugin::getDropdowns()
  src/CommonDropdown.php:115                         Dropdown::getStandardDropdownItemTypes()
  src/Html.php:1455                                  CommonDropdown::getMenuContent()
  src/Html.php:1682                                  Html::generateMenuSession()
  front/central.php:86                               Html::header()
  public/index.php:82                                require()
```

**Comment by ericferon** [comment]
Hi,
Have you also the same symptom ?
Can you access the list of 21 classes via the menu Setup>Dropdowns>Business Processes>Link classes ?
K.r,

**Comment by daufveila** [comment]
Hello, thanks for the answer.
I actually have the same problem. As soon as I activate the plugin, I can no longer load any pages. That's why I can't access the list of 21 classes via the menu.

But I have just discovered something else:

Since I deactivate `php bin/console glpi:plugin:deactivate archibp` the plugin in the cli to restore the functions as before. I also activated `php bin/console glpi:plugin:activate archibp` it via the cli. This time I am able to load most of the pages, but I am still unable to load the list of 21 classes via the menu. Just a blank screen.


Do u know how to load the list of 21 classes via cli?

*EDIT: 
I have also activated `php bin/console glpi:plugin:activate archibp` via the cli. This time I can load most of the pages as Super-Admin, but I still can't load the list of 21 classes via the menu. Just a blank screen.
When I switch from super-Admin to another profile, I get the blank screen again.

**Comment by daufveila** [comment]
After I had activated the plugin via the cli, I tried to call up the configuration in `Setup > Plugin > Configure `  . This is the result:
![image](https://github.com/ericferon/glpi-archibp/assets/49647465/7b39b460-a477-42c1-87ca-70627fbc686a)

I have discovered other things that might help.
When I try to uninstall the plugin to reinstall it, I get this message: 
```
SQL error "1146": table 'glpi.glpi_plugin_archibp_configbplinks' does not exist in query "SELECT `name` FROM `glpi_plugin_archibp_configbplinks` WHERE `name` LIKE 'PluginArchibp%' AND (`as_view_on` IS NULL OR `as_view_on` = '')"
Uncaught exception error: Call of a member function fetch_assoc() on bool in /var/www/glpi/src/DBmysql.php on line 570
```

**Comment by daufveila** [comment]
I got it to "work".
I deleted the installation I did through the marketplace by removing the folder from `rm -r glpi/marktplace/archibp`

Then I downloaded the latest zip from the git project and unpacked it into the plugins folder.

The command 
```bash
php bin/console glpi:plugin:install archibp --force
```
then asked for a user, I chose the glpi user.
```bash
User to use:glpi
```
The installation throws some warnings, which can be seen here.

```bash
Processing plugin "archibp"...
PHP Warning (2): fopen(/var/www/glpi/marketplace/archisw/sql/update-2.0.1.sql): Failed to open stream: No such file or directory in /var/www/glpi/src/DBmysql.php at line 1015
PHP Warning:  fopen(/var/www/glpi/marketplace/archisw/sql/update-2.0.1.sql): Failed to open stream: No such file or directory in /var/www/glpi/src/DBmysql.php on line 1015
PHP Warning (2): Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php at line 66
PHP Warning:  Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php on line 66
PHP Warning (2): Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php at line 67
PHP Warning:  Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php on line 67
PHP User Warning (512): Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php at line 2094
PHP Warning:  Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php on line 2094
PHP User Warning (512): Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_tasktypes`.`id`. in /var/www/glpi/src/DBmysql.php at line 2132
PHP Warning:  Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_tasktypes`.`id`. in /var/www/glpi/src/DBmysql.php on line 2132
PHP Warning (2): Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php at line 66
PHP Warning:  Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php on line 66
PHP Warning (2): Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php at line 67
PHP Warning:  Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php on line 67
PHP User Warning (512): Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php at line 2094
PHP Warning:  Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php on line 2094
PHP User Warning (512): Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_criticities`.`id`. in /var/www/glpi/src/DBmysql.php at line 2132
PHP Warning:  Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_criticities`.`id`. in /var/www/glpi/src/DBmysql.php on line 2132
PHP Warning (2): Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php at line 66
PHP Warning:  Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php on line 66
PHP Warning (2): Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php at line 67
PHP Warning:  Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php on line 67
PHP User Warning (512): Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php at line 2094
PHP Warning:  Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php on line 2094
PHP User Warning (512): Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_tasktargets`.`id`. in /var/www/glpi/src/DBmysql.php at line 2132
PHP Warning:  Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_tasktargets`.`id`. in /var/www/glpi/src/DBmysql.php on line 2132
PHP Warning (2): Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php at line 66
PHP Warning:  Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php on line 66
PHP Warning (2): Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php at line 67
PHP Warning:  Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php on line 67
PHP User Warning (512): Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php at line 2094
PHP Warning:  Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php on line 2094
PHP User Warning (512): Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_taskstates`.`id`. in /var/www/glpi/src/DBmysql.php at line 2132
PHP Warning:  Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_taskstates`.`id`. in /var/www/glpi/src/DBmysql.php on line 2132
PHP Warning (2): Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php at line 66
PHP Warning:  Undefined array key "as_view_on" in /var/www/glpi/plugins/archibp/hook.php on line 66
PHP Warning (2): Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php at line 67
PHP Warning:  Undefined array key "viewlimit" in /var/www/glpi/plugins/archibp/hook.php on line 67
PHP User Warning (512): Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php at line 2094
PHP Warning:  Usage of "utf8" charset/collation detected, should be "utf8mb4" in /var/www/glpi/src/DBmysql.php on line 2094
PHP User Warning (512): Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_swcomponents`.`id`. in /var/www/glpi/src/DBmysql.php at line 2132
PHP Warning:  Usage of signed integers in primary or foreign keys is discouraged, please use unsigned integers instead in `glpi_plugin_archibp_swcomponents`.`id`. in /var/www/glpi/src/DBmysql.php on line 2132
PHP Warning (2): Undefined array key "glpiactiveprofile" in /var/www/glpi/plugins/archibp/inc/profile.class.php at line 235
PHP Warning:  Undefined array key "glpiactiveprofile" in /var/www/glpi/plugins/archibp/inc/profile.class.php on line 235
PHP Warning (2): Trying to access array offset on value of type null in /var/www/glpi/plugins/archibp/inc/profile.class.php at line 235
PHP Warning:  Trying to access array offset on value of type null in /var/www/glpi/plugins/archibp/inc/profile.class.php on line 235
PHP Warning (2): Undefined array key "glpiactiveprofile" in /var/www/glpi/plugins/archibp/hook.php at line 75
PHP Warning:  Undefined array key "glpiactiveprofile" in /var/www/glpi/plugins/archibp/hook.php on line 75
PHP Warning (2): Trying to access array offset on value of type null in /var/www/glpi/plugins/archibp/hook.php at line 75
PHP Warning:  Trying to access array offset on value of type null in /var/www/glpi/plugins/archibp/hook.php on line 75
Plugin "archibp" has been installed and can be activated.
```

But after activation the entry is available for super admins.
```bash
php /var/www/glpi/bin/console glpi:plugin:activate archibp
```
```bash
Processing plugin "archibp"...
Plugin "archibp" has been activated.
```

![image](https://github.com/ericferon/glpi-archibp/assets/49647465/d4da7c4e-67fe-42d7-8153-d4d25ad123e7)


However, there are still limitations. 
In the `Setup>Dropdowns>Business Processes ` menu, I now see the following: 
![image](https://github.com/ericferon/glpi-archibp/assets/49647465/8603c9c6-f0f4-4141-b6d5-1d8a0a5e2660)

But not 21 classes.

And when I try to configure the plugin, I get an error message on the https://glpi.domain.de/plugins/archibp/front/configbp.php page: 
**You do not have permission to perform this action.
Back to the previous page**

I hope the information I have provided will help you in some way.

**Comment by ericferon** [comment]
Hi,
The installation seems OK : the messages are only warnings...
Regarding your access to the configuration, I guess that you need to grant access to the "Configuration" in your used profile (tab "Business Processes" in your profile).
After that operation, you will see "Link classes" in your list of dropdown tables.
K.r,

**Comment by rtfmpliz** [comment]
after follow step by step @daufveila  
I could get my plugin run, but with some error if debug mode is enabled , then could be cleared using 'chown -R user:group *' from /var/www/html/glpi/marketplace/archibp folder
but 21 class not shown up yet
when accees newly created business process in debug mode , there are errors like this 
![image](https://github.com/ericferon/glpi-archibp/assets/9409853/5150bdf8-e569-456b-913a-8e255ecfb714)

but i could access config menu using @ericferon  suggestion
![image](https://github.com/ericferon/glpi-archibp/assets/9409853/3e2435bb-9646-4c23-be88-aa409d1b31af)

**Comment by raregtp** [comment]
Figure this out.....when installed from the Marketplace, the SQL statements are not being ran.  I was also experiencing the exact symptoms, but after running the SQL for a new install of 2.0.1, all database entries were created and the plugin runs normally.  Thanks!!

**Comment by picklesrein** [comment]
Confirming the database isn't being updated:
`Error: Table 'glpi_db.glpi_plugin_archibp_configbplinks' doesn't exist`
so how does one manually run the SQL statements for the plugin installation?

**Comment by picklesrein** [comment]
Resolved it for my situation on 10.0.18-dev. It was necessary for disabling this plugin on the command line and then re-install it via the GUI. 
```
MariaDB [glpi10_db]> select * from glpi_plugins;
+----+---------------+--------------------+---------+-------+------
| id | directory     | name               | version | state | autho
+----+---------------+--------------------+---------+-------+------
|  1 | archibp       | Business Processes | 2.0.11  |     4 | Eric 
|  2 | archisw       | Apps structures    | 3.0.20  |     1 | Eric 
...
+----+---------------+--------------------+---------+-------+------
```
`update glpi_plugins set state = 2 where id=1;`
Then re-install:
![image](https://github.com/user-attachments/assets/5b874d2d-ecc0-4ab1-a165-458e6b432aed)

After that it appears to have the entries it was missing:
```
MariaDB [glpi10_db]> show tables LIKE 'glpi_plugin_archibp%';
+--------------------------------------------+
| Tables_in_glpi10_db (glpi_plugin_archibp%) |
+--------------------------------------------+
| glpi_plugin_archibp_configbpdatatypes      |
| glpi_plugin_archibp_configbpdbfieldtypes   |
| glpi_plugin_archibp_configbpfieldgroups    |
| glpi_plugin_archibp_configbphaligns        |
| glpi_plugin_archibp_configbplinks          |
| glpi_plugin_archibp_configbps              |
| glpi_plugin_archibp_criticities            |
| glpi_plugin_archibp_profiles               |
| glpi_plugin_archibp_swcomponents           |
| glpi_plugin_archibp_tasks                  |
| glpi_plugin_archibp_tasks_items            |
| glpi_plugin_archibp_taskstates             |
| glpi_plugin_archibp_tasktargets            |
| glpi_plugin_archibp_tasktypes              |
+--------------------------------------------+
14 rows in set (0.001 sec)
```

Actually had to perform the above several times as sometimes it returned an incomplete set of tables:
```
MariaDB [glpi10_db]> show tables LIKE 'glpi_plugin_archibp%';
+--------------------------------------------+               
| Tables_in_glpi10_db (glpi_plugin_archibp%) |
+--------------------------------------------+               
| glpi_plugin_archibp_criticities            |
| glpi_plugin_archibp_profiles               |
| glpi_plugin_archibp_swcomponents           |
| glpi_plugin_archibp_tasks                  |
| glpi_plugin_archibp_tasks_items            |
| glpi_plugin_archibp_tasktypes              |
+--------------------------------------------+
6 rows in set (0.001 sec)                     
```
and
```
MariaDB [glpi10_db]> show tables LIKE 'glpi_plugin_archibp%';
+--------------------------------------------+               
| Tables_in_glpi10_db (glpi_plugin_archibp%) |
+--------------------------------------------+
| glpi_plugin_archibp_swcomponents           |
+--------------------------------------------+
1 row in set (0.001 sec)                      
```

**Comment by picklesrein** [comment]
For anyone else reading this, as stated in https://github.com/ericferon/glpi-archibp/issues/4#issuecomment-1855872102, even the super-admin will get errors on configuration until these are checked:

![image](https://github.com/user-attachments/assets/17dbaa40-857b-47d6-b391-21d14af8ac97)
